api/src/main/java/org/apache/iceberg/actions/SnapshotUpdate.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.Snapshot;
+
+/**
+ * An action that produces snapshots. This interface contains common methods for all
+ * actions that create a new table {@link Snapshot}.
+ *
+ * @param <ThisT> the child Java API class, returned by method chaining
+ * @param <R> the Java type of the result produced by this action
+ */
+public interface SnapshotUpdate<ThisT, R> extends Action<R> {
+  /**
+   * Set a summary property in the snapshot produced by this action.
+   *
+   * @param property a String property name
+   * @param value a String property value
+   * @return this for method chaining
+   */
+  ThisT set(String property, String value);
```

Comment:
In other places, we use `with` and descriptive name (option, property). Should this be `withSnapshotProperty`?

Reply:
Good point, this has to be changed.

I am actually no longer sure having `with` prefix is a good idea as we have it for some configs but not others. Shall we consider dropping `with` from names? Then we will have `snapshotProperty`, `tableProperties`, `options`.

```
actions.rewriteManifests()
    .stagingLocation("custom-staging-location")
    .option("use-caching", "true")
    .snapshotProperty("label", "value")
    .execute();
```

```
actions.removeOrphanFiles()
    .location("custom-location")
    .option("use-caching", "true")
    .execute();
```

```
actions.snapshot("db.tbl")
    .as("db.tbl_2")
    .option("use-caching", "true")
    .location("custom-location")
    .tableProperties(props)
    .execute();
```

```
actions.migrate("db.tbl")
    .option("use-caching", "true")
    .tableProperty("p1", "v1")
    .tableProperty("p2", "v2")
    .execute();
```

Reply:
cc @RussellSpitzer who developed the original snapshot/migrate API.

Reply:
I've switched to the new approach.

Reply:
I have no strong opnions about this although I do think we use the preposition-less(?) names with accessors usually rather than setters.